src/routes/tutorials.js

```diff
@@ -0,0 +1,40 @@
+import cache from '../utils/cache.js';
+import { library } from '../utils/kvMetadata.js';
+import notFound from '../utils/notFound.js';
+import respond from '../utils/respond.js';
+
+/**
+ * Register (deprecated) tutorial routes.
+ *
+ * @param {import('hono').Hono} app App instance.
+ */
+export default app => {
+    // Library tutorials
+    app.get('/libraries/:library/tutorials', async ctx => {
+        // Get the library
+        const lib = await library(ctx.req.param('library'), ctx.sentry).catch(err => {
+            if (err.status === 404) return;
```

Comment:
I see the reasoning for pulling this into the library method, but at the same time I also like that those methods _just_ handle the KV-related calls/logic and nothing else, a nice separation of concerns.

Although this is duplicated across the three usages, IMO the correct place is in the handlers as they are what produce the custom response for a 404 from this sub-request API call.

Thoughts?

Reply:
I see what you mean. I am of two minds.

If we know that 404s must **never** raise an error, then I feel it can make sense to abstract that away inside `library`, because we're not deciding on the return status for the caller, we're just swallowing an error that shouldn't be treated as one for callers. Basically, it abstracts the 404 away for the callers, because they shouldn't have to care or know about it. The concern about how to actually handle an error still remains with the caller, so the concerns remain separated.

If there are potential scenarios where 404s should or could be handled differently, then I agree that it makes sense to keep the handling of 404s where they are now.